message.c

```diff
@@ -645,21 +678,32 @@ parse_packet(const unsigned char *from, struct interface *ifp,
             int rc, parsed_len, is_ss;
             if(len < 10) {
                 if(len < 2 || message[3] & 0x80)
-                    have_v4_prefix = have_v6_prefix = 0;
+                    have_v4_prefix = have_v6_prefix = have_v4ov6_prefix = 0;
                 goto fail;
             }
+            if(!ae_exists(message[2])) {
+                debugf("Received update with unknown AE %d. Ignoring.\n",
+                       message[2]);
+                goto done;
+            }
             DO_NTOHS(interval, message + 6);
             DO_NTOHS(seqno, message + 8);
             DO_NTOHS(metric, message + 10);
             if(message[5] == 0 ||
-               (message[2] == 1 ? have_v4_prefix : have_v6_prefix))
+               (message[2] == AE_IPV4 ? have_v4_prefix :
+                message[2] == AE_IPV6 ? have_v6_prefix :
+                message[2] == AE_V4OV6 ? have_v4ov6_prefix :
+                0))
                 rc = network_prefix(message[2], message[4], message[5],
                                     message + 12,
-                                    message[2] == 1 ? v4_prefix : v6_prefix,
+                                    (message[2] == AE_IPV4 ? v4_prefix :
```

Comment:
Why use parentheses surrounding the ternary expression?

Reply:
Mostly for visual grouping, as this spans on multiple lines. If you'd rather not have those, I could remove it.